File: app/javascript/channels/tweets_channel.js
import consumer from "channels/consumer"

consumer.subscriptions.create("TweetsChannel", {
  connected() {
    // Called when the subscription is ready for use on the server
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    // Called when there's incoming data on the websocket for this channel
    const event = new CustomEvent("newTweet", { detail: data });
    window.dispatchEvent(event);
  }
});

File: app/channels/tweets_channel.rb
class TweetsChannel < ApplicationCable::Channel
  def subscribed
    # stream_from "some_channel"
    stream_from "tweets_channel"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def receive(data)
    ActionCable.server.broadcast('tweets_channel', data)
  end
end
